src/main.js

```diff
@@ -22,45 +22,55 @@ const ipc = electron.ipcMain;
  * */
 let mainWindow;
 
+/** NT IDS of strings  */
+let stringIDs = [];
 
-let connected, ready = false;
+let connected,
+    ready = false;
 function createWindow() {
     // Attempt to connect to the localhost
     client.start((con, err) => {
         // If the Window is ready than send the connection status to it
         if (ready) {
-            mainWindow.webContents.send('connected', con);
-        }
-        else
-            connected = () => mainWindow.webContents.send('connected', con);
+            mainWindow.webContents.send("connected", con);
+        } else connected = () => mainWindow.webContents.send("connected", con);
     });
     // When the script starts running in the window set the ready variable
-    ipc.on('ready', (ev, mesg) => {
+    ipc.on("ready", (ev, mesg) => {
         ready = true;
         // Send connection message to the window if if the message is ready
-        if (connected)
-            connected();
+        if (connected) connected();
     });
     // When the user chooses the address of the bot than try to connect
-    ipc.on('connect', (ev, address, port) => {
+    ipc.on("connect", (ev, address, port) => {
         let callback = (connected, err) => {
-            mainWindow.webContents.send('connected', connected);
+            mainWindow.webContents.send("connected", connected);
         };
         if (port) {
             client.start(callback, address, port);
-        }
-        else {
+        } else {
             client.start(callback, address);
         }
     });
-    ipc.on('add', (ev, mesg) => {
+    ipc.on("add", (ev, mesg) => {
         client.Assign(mesg.val, mesg.key, (mesg.flags & 1) === 1);
     });
-    ipc.on('update', (ev, mesg) => {
+    ipc.on("update", (ev, mesg) => {
+        if (stringIDs.indexOf(mesg.id) >= 0) {
```

Comment:
What's this doing?

Reply:
Users who receive a boolean will most like send a boolean back, which will then needs to be reverted back to a string since the NT server does not allow type changes